crates/core/src/error.rs

```diff
@@ -790,3 +797,83 @@ where
         Self(RusticErrorKind::from(value))
     }
 }
+
+#[non_exhaustive]
+#[derive(Error, Debug, Display)]
+pub enum CheckError {
+    /// error reading pack {id} : {err}
+    ErrorReadingPack { id: Id, err: RusticError },
```

Comment:
```suggestion
    ErrorReadingPack { id: Id, err: Box<RusticError> },
```

Reply:
I assume it makes sense to also box this one here, as the RusticError became quite large over time